tests/sudorule/test_sudorule.yml

```diff
@@ -0,0 +1,286 @@
+    failed_when: not result.changed
+
+  - name: Ensure sudorule is present, with hostategory 'all', again
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: allhosts
+      hostcategory: all
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure sudorule is disabled
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      state: disabled
+
+  - name: Ensure sudorule is disabled, again
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      state: disabled
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure sudorule is enabled
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      state: enabled
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure sudorule is enabled, again
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      state: enabled
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure sudorule is present and some sudocmd are a member of it.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      cmd:
+      - /sbin/ifconfig
+      - /usr/bin/vim
+      action: member
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure sudorule is present and some sudocmd are a member of it, again.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      cmd:
+      - /sbin/ifconfig
+      - /usr/bin/vim
+      action: member
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure sudorule is present with cmdcategory 'all'.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: allcommands
+      cmdcategory: all
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure sudorule is present with cmdcategory 'all', again.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: allcommands
+      cmdcategory: all
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure host "{{ groups.ipaserver[0] }}" is present in sudorule.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      host: "{{ groups.ipaserver[0] }}"
+      action: member
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure host "{{ groups.ipaserver[0] }}" is present in sudorule, again.
+    ipasudorule:
+      ipaadmin_password: pass1234
+      name: testrule1
+      host: "{{ groups.ipaserver[0] }}"
+      action: member
+    register: result
+    failed_when: result.changed
+
+  - name: Ensure hostgroup is present in sudorule.
+    ipasudorule:
+      ipaadmin_password: MyPassword123
+      name: testrule1
+      hostgroup: cluster
+      action: member
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure hostgroup is present in sudorule, again.
+    ipasudorule:
+      ipaadmin_password: MyPassword123
+      name: testrule1
+      host: server
```

Comment:
The test is failing as `host: server` has not been added before to `testrule1` before.
Should this be `hostgroup: cluster`?

Reply:
Fixed.